File: src/ts/models/item.ts
import {Status} from "./status"
import {Disposition} from "./disposition"
export class Item {
 id: string = "";
 agentId: string|null = null;
 boxNo: string|null = null;
 userId: string|null = null;
 model: string|null = null;
 serialNo:string|null = null;
 macNo1:string|null = null;
 macNo2:string|null = null;
 note:string|null = null;
 quantity:number  = 0;
 purchasePrice:number  = 0;
 retailPrice:number  = 0;
 location:string|null = null;
 symptom:string|null = null;
 invoiceAt: string|null = null;
 soldAt: string|null = null;
 shippedAt: string|null = null;
 createdAt: string|null = null;
 updatedAt:string|null = null;
 status: Status | null = null;
 dispositions: Disposition[] | null = null;

}

File: src/Entities/Item.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Intertel.Entities
{
    public class Item
    {
        [Key]
        public Guid Id { get; set; }
        public string UserId { get; set; }
        public string AgentId { get; set; }
        public string Model { get; set; }
        public string SerialNo { get; set; }
        public string MacNo1 { get; set; }
        public string MacNo2 { get; set; }
        public string BoxNo { get; set; }
        public int Quantity { get; set; }
        public string Location { get; set; }
        public decimal RetailPrice { get; set; }
        public decimal PurchasePrice { get; set; }
        public string Symptom { get; set; }
        public string Purpose { get; set; }
        public string Note { get; set; }
        public DateTime? InvoiceAt { get; set; } = null;
        public DateTime? SoldAt { get; set; } = null;
        public DateTime? ShippedAt { get; set; } = null;
        public DateTime? ReturnedAt { get; set; } = null;

        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        public DateTime? CreatedAt { get; set; } = DateTime.Now;
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        public DateTime? UpdatedAt { get; set; } = DateTime.Now;
        public Status Status { get; set; }
        public virtual ICollection<Disposition> Dispositions { get; set; }
        public Customer Customer { get; set; }

    }
}